app/Jobs/ReportSpam.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\User;
+use App\Notifications\MarkedAsSpamNotification;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Notification;
+
+class ReportSpam implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    public function __construct(private User $user, private Model $spammable)
```

Comment:
Can we create a `Spammable` interface here instead of referencing the Eloquent Model class? Just like we do for ReplyAble, etc.

Reply:
I've committed some changes to accomodate this, however it will require the `Reply` and `Thread` class to be NOT final... what do you think?